### ISSUE TITLE
JSON appears in Error List's Code filter menu popup

### ISSUE THREAD

**Issue by Jehoel** [dotnet/roslyn#52839]
### Analyzer

Unknown
**UPDATE:** It seems to be IntelliCode.

### Analyzer source

Unsure.

### Describe the bug

After using Visual Studio for more than an hour or so, hidden code analysis errors start to amass in the Error List's Code column filter popup, they appear as serialized JSON with zero occurrence counts. 

![image](https://user-images.githubusercontent.com/1693078/115474692-d608f180-a1f2-11eb-932d-d279805c841d.png)

An example entry is:

```
{"NewText": "Action ?", "EditRange": {"Item1":49,"Item2":32, "EndLocation":{"Item1":49,"Item2":38}},"ProgramIdentifier":"-1849195906_-1851706466", "EditType":0}
```

### Steps To Reproduce

1. Create a .NET Class Library project that multi-targets .NET Framework, .NET Core 3.1, and .NET Standard 2.0 and enable nullable reference types and C# 8.0 or later.
2. Add C# code that contains triggers static analysis warnings.
3. Spend an hour or so working on the C# code to address the prior warnings.
4. Open the Error List and open the Code column filter and you should at least see a few of these entries.

### Expected behavior

Them to not appear.

### Actual behavior

They appear.

## Additional context

Visual Studio 2019 16.9.4

**Comment by mavasani** [comment]
Tagging @tmat - are these EnC diagnostics?

**Comment by tmat** [comment]
I don't know what these are.

**Comment by Jehoel** [comment]
FWIW, I ran `strings -n 17 -s * | findstr /i ProgramIdentifier` on my VS install and got this:

![image](https://user-images.githubusercontent.com/1693078/115892056-a5e86b00-a40b-11eb-8a18-c26af0b388ea.png)

```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.ProgramSynthesis.Suggestions.Code.Client.dll: <ProgramIdentifier>k__BackingField
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.ProgramSynthesis.Suggestions.Code.Client.dll: get_ProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.ProgramSynthesis.Suggestions.Code.Client.dll: set_ProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.ProgramSynthesis.Suggestions.Code.Engine.dll: <SubsumingProgramIdentifier>k__BackingField
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.ProgramSynthesis.Suggestions.Code.Engine.dll: <TransientProgramIdentifier>k__BackingField
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.ProgramSynthesis.Suggestions.Code.Engine.dll: get_ProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.ProgramSynthesis.Suggestions.Code.Engine.dll: set_ProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.ProgramSynthesis.Suggestions.Code.Engine.dll: get_SubsumingProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.ProgramSynthesis.Suggestions.Code.Engine.dll: subsumingProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.ProgramSynthesis.Suggestions.Code.Engine.dll: suggestionProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.ProgramSynthesis.Suggestions.Code.Engine.dll: GetProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.ProgramSynthesis.Suggestions.Code.Engine.dll: get_TransientProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.ProgramSynthesis.Suggestions.Code.Engine.dll: transientProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.ProgramSynthesis.Suggestions.Code.Engine.dll: programIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.dll: <ProgramIdentifier>k__BackingField
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.dll: <programIdentifier>k__BackingField
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.dll: get_ProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.dll: set_ProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.dll: getTextEditsForProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.dll: getNextTextEditsForProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.dll: getSuggestionCountForProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.dll: reportedErrorCodeByProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.dll: ReportedRecursionByProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.dll: get_programIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.dll: set_programIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.dll: VS.IntelliCode.RepeatedEdits.Suggestion.ProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.dll: VS.IntelliCode.RepeatedEdits.Suggestion.ProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.Refactorings.dll: <GetEditSuggestionsByProgramIdentifier>b__0
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.Refactorings.dll: <GetEditSuggestionsByProgramIdentifier>b__1
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.Refactorings.dll: get_ProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.Refactorings.dll: lastAppliedProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.Refactorings.dll: getTextEditsForProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.Refactorings.dll: getNextTextEditsForProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.Refactorings.dll: getSuggestionCountForProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.Refactorings.dll: GetProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.Refactorings.dll: GetEditSuggestionsByProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.Refactorings.dll: programIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.Refactorings.dll: VS.IntelliCode.RepeatedEdits.Suggestion.ProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.Refactorings.dll: VS.IntelliCode.RepeatedEdits.Suggestion.ProgramIdentifier
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Microsoft\IntelliCode\Microsoft.VisualStudio.IntelliCode.Refactorings.dll: newProgramIdentifiers
```

My money's on IntelliCode being responsible - IntelliCode has been the source of _no end_ of bugs, glitches, and performance issues with Visual Studio, and most infuriatingly many other features have started to have a hard-dependency on IntelliCode so it's not like I can simply disable it.

**Comment by jmarolf** [comment]
@Jehoel good analysis, I agree this looks like intellicode. If you turn off all intellicode features in tools->options I assume this goes away?

**Comment by Jehoel** [comment]
@jmarolf  Yeah, I just disabled IntelliCode (and LiveShare, and Rich Navigation, just to be sure) and my Error List's filters no-longer have those items. And VS performance is MUCH higher too. I noticed that IntelliCode's disk logging feature seems to be a bottleneck in many cases (as though it's doing synchronous disk IO and blocking the editor...)

**Comment by jinujoseph** [comment]
Routed to [intellicode ](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1320713)team
